akka-actor-tests/src/test/scala/akka/io/InetAddressDnsResolverSpec.scala

```diff
@@ -0,0 +1,100 @@
+package akka.io
+
+import java.security.Security
+import java.util.concurrent.TimeUnit
+
+import akka.actor.Props
+import akka.testkit.{ AkkaSpec, TestActorRef }
+
+class InetAddressDnsResolverSpec extends AkkaSpec("""
+    akka.io.dns.inet-address.positive-ttl = default
+    akka.io.dns.inet-address.negative-ttl = default
+    akka.actor.serialize-creators = on
+    """) { thisSpecs ⇒
+
+  "The DNS resolver default ttl's" must {
+    "use the default value for positive caching if it is not overridden" in {
+      withNewSecurityProperty("networkaddress.cache.ttl", "") {
+        withNewSystemProperty("sun.net.inetaddr.ttl", "") {
+          dnsResolver.positiveTtl shouldBe secondsToMillis(30)
```

Comment:
could fail depending on JVM version though, right?

Reply:
I don't think so. Although the assumption of the test is based on the Java 8 implementation, the Java 9 version remains with the same logic (although implementation has changed a little) 

http://hg.openjdk.java.net/jdk9/jdk9/jdk/file/19cbbf152335/src/share/classes/sun/net/InetAddressCachePolicy.java